plugins/modules/nomad_token.py

```diff
@@ -0,0 +1,285 @@
+            use_ssl=dict(type='bool', default=True),
+            timeout=dict(type='int', default=5),
+            validate_certs=dict(type='bool', default=True),
+            client_cert=dict(type='path'),
+            client_key=dict(type='path'),
+            namespace=dict(type='str'),
+            token=dict(type='str', no_log=True),
+            name=dict(type='str'),
+            token_type=dict(choices=['client', 'management', 'bootstrap'], default='client'),
+            policies=dict(type='list', elements='str', default=[]),
+            global_replicated=dict(type='bool', default=False),
+        ),
+        supports_check_mode=True,
+        required_one_of=[
+            ['name', 'token_type']
+        ]
+    )
+
+    if not import_nomad:
+        module.fail_json(msg=missing_required_lib("python-nomad"))
+
+    certificate_ssl = (module.params.get('client_cert'), module.params.get('client_key'))
+
+    nomad_client = nomad.Nomad(
+        host=module.params.get('host'),
+        port=module.params.get('port'),
+        secure=module.params.get('use_ssl'),
+        timeout=module.params.get('timeout'),
+        verify=module.params.get('validate_certs'),
+        cert=certificate_ssl,
+        namespace=module.params.get('namespace'),
+        token=module.params.get('token')
+    )
+
+    changed = False
+    result = {}
+    if module.params.get('state') == "present":
+
+        if module.params.get('token_type') == 'bootstrap':
+            try:
+                current_token = get_token('Bootstrap Token', nomad_client)
+                if current_token:
+                    result = {"msg": "ACL bootstrap already exist.", "token": {}}
+
+            except nomad.api.exceptions.URLNotAuthorizedNomadException:
+                try:
+                    nomad_result = nomad_client.acl.generate_bootstrap()
+                    result = {'msg': "Boostrap token created.",
+                              "token": transform_response(nomad_result)}
+                    changed = True
+                except Exception as e:
+                    module.fail_json(msg=to_native(e))
+        else:
+            try:
+                token_info = {
+                    "Name": module.params.get('name'),
+                    "Type": module.params.get('token_type'),
+                    "Policies": module.params.get('policies'),
+                    "Global": module.params.get('global_replicated')
+                }
+
+                current_token = get_token(token_info['Name'], nomad_client)
+
+                if current_token:
+                    token_info['AccessorID'] = current_token['AccessorID']
+                    nomad_result = nomad_client.acl.update_token(current_token['AccessorID'],
+                                                                 token_info)
+                    result = {"msg": "Acl token updated.",
+                              "token": transform_response(nomad_result)}
```

Comment:
How does the above code determine whether something was actually changed?

Reply:
It really is a mistake. Thanks, I'll fix it